internal/target/repository_credential_library.go

```diff
@@ -0,0 +1,349 @@
+package target
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+)
+
+// AddTargetCredentialLibraries adds the clIds to the targetId in the repository. The target
+// and the list pf credential libraries attached to the target, after clIds are added,
+// will be returned on success.
+// The targetVersion must match the current version of the targetId in the repository.
+func (r *Repository) AddTargetCredentialLibraries(ctx context.Context, targetId string, targetVersion uint32, clIds []string, _ ...Option) (Target, []*CredentialLibrary, error) {
```

Comment:
The successful result of any operation by the API is to return the resource itself (including the host sets).  Can you add the []TargetSets as a return value (and add []*CredentialLibrary for the SAD HostSet operations?)

Reply:
Great point!